### ISSUE TITLE
Allow goavro version to appear in go.mod (change module path to include `/v2`)

### ISSUE THREAD

**Issue by smyrman** [linkedin/goavro#159]
I noticed a few things with `go mod` and `goavro` that appear to not be working ideally.

1. When installing with `go get -u github.com/linkedin/goavro`, `v2.1.0` is installed (marked as latest version on GitHub). Is this intentional? Appears related to #149 and #143.
2. When specifically installing a later version, the version don't appear in `go.mod`, but get's a v0.0.0 pseudo-version.
3. go get with `/v2` suffix in import path don't work.

## Steps to reproduce
Works, but doesn't record the version:
```
go get -u github.com/linkedin/goavro@v2.8.0
```

## Results in `go.mod`
```
+github.com/linkedin/goavro v0.0.0-20190425224415-ce0a001c0b8c h1:jyaiBKnmWa+l7dPQyHCFpNBdYoZZ8mP/4zyz5Wk1KN4=
+github.com/linkedin/goavro v0.0.0-20190425224415-ce0a001c0b8c/go.mod h1:dCvY+Sak0ZYfzkaf551EqsVzVWShWpePCjWbSBtdxqM=
```

## Expected

Preferred go get commands (but neither currently work):
```
go get -u github.com/linkedin/goavro/v2
```
```
go get -u github.com/linkedin/goavro/v2@v2.8.0
```
Resulting in `go.mod` entries:
```
 github.com/linkedin/goavro/v2 v2.8.0 HASH
 github.com/linkedin/goavro/v2 v2.8.0 HASH
```

# Solution

From what I can understand, there is _non need_ to alter the file sturcture of the repo, but the `go.mod` file should be updated to append `/v2` to the import path.

Exmample of a repo that appear to work with this set-up:
- https://github.com/justinfx/gofileseq

There is no `v2` folder, but `go.mod` states `module github.com/justinfx/gofileseq/v2`. As a consequence, this works fine:

```sh
$ go get github.com/justinfx/gofileseq/v2                                                                                   :(
go: finding github.com/justinfx/gofileseq/v2 v2.6.3
go: finding github.com/jessevdk/go-flags v0.0.0-20131218111322-7168dbf82f26
go: finding github.com/MichaelTJones/walk v0.0.0-20140707193007-3af09438b0ab
go: downloading github.com/justinfx/gofileseq/v2 v2.6.3
go: extracting github.com/justinfx/gofileseq/v2 v2.6.3
```

So the fix for `goavro` should be to just change the module path in `go.mod` as well to `github.com/linkedin/goavro/v2`

**Comment by smyrman** [comment]
@y0ssar1an, in case you are interested.

**Comment by karrick** [comment]
@smyrman, thanks for the link and example to the other library. For a long time I have pondered how I would solve this particular problem. Giving it a shot now.

**Comment by karrick** [comment]
Fixed by a1a32f3
